src/snowflake/snowpark/session.py

```diff
@@ -2276,6 +2303,30 @@ def create_dataframe(
                 )
                 sf_schema = self._conn._get_current_parameter("schema", quoted=False)
 
+                # If the user specifies schema for their dataframe, we try out best to match
+                # it by create a temp table with the specified schema, and load the data into
+                # the temp table. If we fail, go back to old method using infer schema.
+                if isinstance(
+                    schema, StructType
+                ) and self._create_temp_table_for_given_schema(temp_table_name, schema):
+                    try:
+                        t = self.write_pandas(
+                            data,
+                            temp_table_name,
+                            database=sf_database,
+                            schema=sf_schema,
```

Comment:
I'm wondering if this is still considered a BCR -- let's say previously even users specify the schema, but it's not appreciated when input is a pandas DF, so the schema of the result DF is different than what users specified.

however, with the change, it can be the schema users specified.